### ISSUE TITLE
Allow manual cache key expiration

### ISSUE THREAD

**Issue by anna-geller** [PrefectHQ/prefect#7315]
### First check

- [X] I added a descriptive title to this issue.
- [X] I used the GitHub search to find a similar request and didn't find it.
- [X] I searched the Prefect documentation for this feature.

### Prefect Version

2.x

### Describe the current behavior

Currently, users can set an expiration time on a cache key 

However, there are circumstances when users want to manually clear cache, possibly before the planned expiration date 

### Describe the proposed behavior

- Add a utility function allowing to manually clear cache for a given task run ID (or possibly flow run ID once/if we add flow-level caching)
- Add a route allowing to do the same via an API call
- Add a capability to do that from the UI (most likely a follow-up feature we'll create for the UI team once backend work is done)

**Comment by tobias-leo** [comment]
Hello @anna-geller, 

We are currently exploring caching for our flows and just ran into a similar situation. Great to see that there is already an open issue.

We want to be able to force-execute tasks in different situations: 
- We want to be able to use a boolean Flow parameter to control whether cached results are used (default false, set by the operator via UI).
- If we execute task T from flow A, using the cached results is acceptable. If we execute task T from flow B, it should not use the cache.

What force-execute means: Found this [code snippet](https://github.com/PrefectHQ/prefect/blob/240c8c45a9a4cd46112bc3b90ac931c7e605cd19/src/prefect/orion/orchestration/core_policy.py#L212 ), perhaps it is only about overriding the decision made here.
- Already cached results for a given cache key are **not** used, even if they are not expired.
- After the execution of the task, the newly computed results are cached.

Adding a task parameter/decorator "cache_force_execute" may conveniently solve this.

Advantages over just the API feature request from my perspective: 
- I only invalidate cached results for a subset of inputs, only the ones I force-execute the task for.
- To invalidate cached results for a given task, I do not have to write a utility function called before the task execution or ask the operator to do it manually.


What do you think?

**Comment by anna-geller** [comment]
> We want to be able to use a boolean Flow parameter to control whether cached results are used
it's essentially how caching already works: if your cache key function returns the same key, a cache is used, if it returns a different key, task run is computed without cache

I suggest reexamining your cache key function - your use case is already possible

**Comment by discdiver** [comment]
@billpalombi mentioned we might also want to have the option to clear the cache per deployment or per workspace.

**Comment by tobias-leo** [comment]
We see a great desire from the operations/business teams to make them able to ignore cached results. We just had a presentation with our operations team regarding the introduction of caching for our flows. The first question was: "Are we able to force execute flows?". 

I want to point out the behavior of such a force-run of a flow:
1. All tasks are executed, regardless of whether there are cached results.
2. The computed return of each task is cached again.
- overwriting the decision taken here: [core_policy.py/CacheRetrieval](https://github.com/PrefectHQ/prefect/blob/240c8c45a9a4cd46112bc3b90ac931c7e605cd19/src/prefect/orion/orchestration/core_policy.py#L212)

Changing the cache key based on a "force_execution" input parameter (e.g., by adding a timestamp or random number) will put the results into the cache with a cache key that will not hit by the next run.
-> Having an API call for clearing the cache per cache key would allow us to do it ourselves upfront.

A typical task in our setting looks like this:
- Input: S3 prefix to a huge image file (S3), some configuration for processing.
- Output: S3 prefix to a huge image file (S3) written by the task.

One straightforward scenario where a force re-execution is beneficial: the output file got corrupted. -> currently we are able to rewrite it only by removing the previous flow runs executed with the same parameters via UI.
We already have a custom cache key function for the input file: The cache key takes into account if the input file has changed (based on ETag or last_modified).

**Comment by trahloff** [comment]
Hi @billpalombi, I was just revisiting this topic because we hit a case where we automatically want to expire specific cached results. 

This issue is marked as completed because of https://github.com/PrefectHQ/prefect/pull/7856, correct? 

From my perspective, the `refresh_cache` option already provides a lot of value but does not fully meet the "proposed behaviour" that Anna outlined in her initial feature request:

- Add a utility function allowing to manually clear cache for a given task run ID (or possibly flow run ID once/if we add flow-level caching)
- Add a route allowing to do the same via an API call
- Add a capability to do that from the UI (most likely a follow-up feature we'll create for the UI team once backend work is done)

The `refresh_cache` option allows you to force execute a flow run and "ignore" the cache.

Proposed capability in addition to Anna's description: Allow users to specify a "time-to-live" value when configuring the cache. 
This is often present in caching solutions and sometimes even key-value-stores (e.g.: DynamoDB) and would allow for automated cache expiration. 

Example business value proposition: I analyze daily changing prices via Prefect. I get the price information via the API "GET https://prices.com/fluffy-unicorns". Currently, I cannot cache this with Prefect because it is always the same API resource and only the return value changes. 
If the user could specify a TTL of 24h, that would be possible. Additionally, you win the feature of an auto-cleaned Prefect caching location. 

What do you think? Does it make sense and if yes, would you like to have this in a separate ticket or reactive this one?
